crates/stackable-operator/src/builder/pod/container.rs

```diff
@@ -188,29 +208,53 @@ impl ContainerBuilder {
         self
     }
 
+    /// This function only adds the [`VolumeMount`] in case there is no volumeMount with the same mountPath already.
+    /// In case there already was a volumeMount with the same path already, an [`tracing::error`] is raised in case the
+    /// contents of the volumeMounts differ.
+    ///
+    /// Historically this function unconditionally added volumeMounts, which resulted in invalid
+    /// [`k8s_openapi::api::core::v1::PodSpec`]s, as volumeMounts where added multiple times - think of Trino using the same [`crate::commons::s3::S3Connection`]
+    /// two times, resulting in e.g. the s3 credentials being mounted twice as the same volumeMount.
```

Comment:
suggestion: Rewording

```suggestion
    /// Adds a new [`VolumeMount`] to the container while ensuring that no
    /// duplicate exists.
    ///
    /// This function ensures that no two [`VolumeMount`]s with the same mount
    /// path can be added. An appropriate error is returned when such a duplicate
    /// mount path is encountered.
    ///
    /// ### Note
    ///
    /// Previously, this function unconditionally added [`VolumeMount`]s, which
    /// resulted in invalid [`PodSpec`][1]s.
    ///
    /// [1]: k8s_openapi::api::core::v1::PodSpec
```

Reply:
Actually it depends on the interpretation of duplicates. For me duplicates are totally fine.
WDYT?
```
    /// Adds a new [`VolumeMount`] to the container while ensuring that no
    /// clashing [`VolumeMount`] exists.
    ///
    /// A clashing [`VolumeMount`] would have same mountPath but a different
    /// content than another [`VolumeMount`]. An appropriate error is returned
    /// when such a clashing mount path is encountered.
    ///
    /// ### Note
    ///
    /// Previously, this function unconditionally added [`VolumeMount`]s, which
    /// resulted in invalid [`PodSpec`][1]s.
    ///
    /// [1]: k8s_openapi::api::core::v1::PodSpec
    ```

Reply:
For me, two or more volume mounts are duplicated when the mount paths are identical, because it is effectively the same volume mount. The same mount path can only be referenced at most once per container.

Reply:
For me mounts are duplicated when they are the same (so same everything, as `==` would check.) And it's totally fine to have duplicates (as the same ca crt can be added multiple times, e.g. because you have 5 identical S3Connection references), the problems start when we have two different mounts (e.g. pulling in different volumes) sharing the same mountPath.

> Adds a new [`VolumeMount`] to the container while ensuring that no duplicate exists.

I would disagree and say it's totally fine to have duplicates. But it's not okay to have clashing mounts (same path, different content)

Reply:
So I feel like we are talking about the same thing here. A duplicate mount is a mount which uses the _same_ mount path.

Mounts which are identical (so `==`) are still duplicates, but more specifically copies of each other.

I'm fine with your proposed doc comment if you find the usage of the word "clashing" more suitable then "duplicate".

Reply:
I think clashing describes it better than duplicate and would pick the proposed doc comment. @NickLarsenNZ please feel free to object

Reply:
Implemented in a sync meeting